src/components/VideoLibraryModal.vue

```diff
@@ -351,100 +673,184 @@ const deselectAllVideos = (): void => {
+    isMultipleSelectionMode.value = true
+  } else {
+    snackbarMessage.value = 'No processed videos found'
+    openSnackbar.value = true
+  }
+}
+
 // Add the log files to the list of files to be downloaded/discarded
-const addSubtitlesToFileList = (filenames: string[]): string[] => {
-  const filesWithSubtitles = filenames.flatMap((filename) => {
-    const subtitleFilename = filename.replace('.webm', '.ass')
-    // Check if the .ass file exists in availableVideos
-    const subtitleExists = availableLogFiles.value.some((video) => video.filename === subtitleFilename)
-    return subtitleExists ? [filename, subtitleFilename] : [filename]
+const addLogDataToFileList = (fileNames: string[]): string[] => {
+  const filesWithLogData = fileNames.flatMap((fileName) => {
+    const subtitlefileName = fileName.replace('.webm', '.ass')
+    const subtitleExists = availableLogFiles.value.some((video) => video.fileName === subtitlefileName)
+    return subtitleExists ? [fileName, subtitlefileName] : [fileName]
   })
-  return filesWithSubtitles
+  return filesWithLogData
 }
 
-const downloadAndUpdateDB = async (filenames: string[]): Promise<void> => {
-  snackbarMessage.value = 'Preparing download...'
+const downloadVideoAndTelemetryFiles = async (): Promise<void> => {
+  snackbarMessage.value = 'Getting your download ready...'
+  let tempProcessedVideos: string[] = []
+  let tempUnprocessedVideos: string[] = []
 
-  await videoStore.downloadFilesFromVideoDB(addSubtitlesToFileList(filenames))
+  selectedVideos.value.forEach((video) => {
+    if (video.isProcessed) tempProcessedVideos.push(video.fileName)
+    if (!video.isProcessed && video.hash) tempUnprocessedVideos.push(video.hash)
+  })
   openSnackbar.value = true
+  if (tempProcessedVideos.length > 0)
+    await videoStore.downloadFilesFromVideoDB(addLogDataToFileList(tempProcessedVideos))
+  if (tempUnprocessedVideos.length > 0) await videoStore.downloadTempVideo(tempUnprocessedVideos)
 }
 
-const discardAndUpdateDB = async (filenames: string[]): Promise<void> => {
+const discardVideosAndUpdateDB = async (): Promise<void> => {
   let selectedVideoArraySize = selectedVideos.value.length
-  snackbarMessage.value = `${selectedVideoArraySize} video(s) discarded.`
+  let processedVideosToDiscard: string[] = []
+  let unprocessedVideosToDiscard: string[] = []
 
-  await videoStore.discardFilesFromVideoDB(addSubtitlesToFileList(filenames))
+  await selectedVideos.value.forEach((video: VideoLibraryFile) => {
+    if (video.isProcessed) processedVideosToDiscard.push(video.fileName)
+    if (!video.isProcessed && video.hash) unprocessedVideosToDiscard.push(video.hash)
+  })
+  processedVideosToDiscard.length > 0 &&
+    (await videoStore.discardProcessedVideosFromDB(addLogDataToFileList(processedVideosToDiscard)))
+  unprocessedVideosToDiscard.length > 0 && (await videoStore.discardUnprocessedVideosFromDB(unprocessedVideosToDiscard))
+
```

Comment:
I also like using `&&` when everything fits in one line and the methods are clear, but here I think they are making reading the code a lot harder. Suggestion to improve code readability:

```typescript
const dataLogFiles = addLogDataToFileList(processedVideosToDiscard)

if (processedVideosToDiscard.length > 0) {
  await videoStore.discardProcessedVideosFromDB(dataLogFiles)
}

if (unprocessedVideosToDiscard.length > 0) {
  await videoStore.discardUnprocessedVideosFromDB(unprocessedVideosToDiscard)
}
```